python/pyspark/serializers.py

```diff
@@ -356,6 +356,33 @@ def __repr__(self):
         return "ArrowStreamPandasSerializer"
 
 
+class InterleavedArrowReader(object):
```

Comment:
I wonder if it's better to have this be a inner class of PandasCogroupSerializer. The reason being this seems tightly coupled with the cogroup deserializer and probably not too useful by itself? 

@BryanCutler @HyukjinKwon What do you think?

Reply:
after @BryanCutler's suggestion we no longer need this.